calcite-qml/Calcite/RadioDelegate.qml

```diff
@@ -15,34 +16,62 @@
  ******************************************************************************/
 import QtQuick 2.15
 import QtQuick.Templates 2.15 as T
+import QtGraphicalEffects 1.15
 
 T.RadioDelegate {
     id: control
 
-    implicitWidth: Math.max(background ? background.implicitWidth : 0,
-                            contentItem.implicitWidth + leftPadding + rightPadding)
-    implicitHeight: Math.max(background ? background.implicitHeight : 0,
-                             Math.max(contentItem.implicitHeight,
-                                      indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
+    Component.onCompleted: console.log(width)
+    implicitWidth: Math.max(
+                       background ? background.implicitWidth : 0,
+                       contentItem.implicitWidth + leftPadding + rightPadding)
+    implicitHeight: Math.max(
+                        background ? background.implicitHeight : 0, Math.max(
+                            contentItem.implicitHeight,
+                            indicator ? indicator.implicitHeight : 0) + topPadding + bottomPadding)
 
     padding: 5
     spacing: 5
 
+    icon.width: 24
+    icon.height: 24
+    icon.source: "images/bullet-point.svg"
+    font.bold: control.checked || control.highlighted
+
     contentItem: Text {
-        leftPadding: control.mirrored ? control.indicator.width + control.spacing : 0
-        rightPadding: !control.mirrored ? control.indicator.width + control.spacing : 0
+        leftPadding: !control.mirrored ? control.indicator.width + control.spacing : 0
+        rightPadding: control.mirrored ? control.indicator.width + control.spacing : 0
         text: control.text
         font: control.font
         color: control.enabled ? Calcite.text1 : Calcite.text3
         elide: Text.ElideRight
         visible: control.text
-        horizontalAlignment: Text.AlignHCenter
+        horizontalAlignment: Text.AlignLeft
         verticalAlignment: Text.AlignVCenter
     }
 
-    indicator: RadioIndicator {
-        x: control.mirrored ? control.leftPadding : control.width - width - control.rightPadding
+    indicator: Image {
+        id: indicator
+        visible: control.checked || control.highlighted || control.hovered
+        x: control.mirrored ? control.width - width - control.rightPadding : control.leftPadding
         y: control.topPadding + (control.availableHeight - height) / 2
-        control: control
+        source: icon.source
+        sourceSize: Qt.size(icon.width, icon.height)
+        width: icon.width
+        height: icon.height
+        ColorOverlay {
+            anchors.fill: indicator
+            source: indicator
+            color: control.checked
+                   || control.highlighted ? Calcite.brand : control.hovered ? Calcite.border1 : "transparent"
+            visible: indicator.visible
+        }
+    }
```

Comment:
🔧  Other calcite controls change the opacity when disabled , you might want to test checking/unchecking the enabled box in the calcite demo app to ensure the component "looks" disabled when required.

<img width="172" alt="image" src="https://user-images.githubusercontent.com/42203545/153211784-d806ba4a-be54-4853-8c1a-4cb878d6529e.png">